boot/zephyr/main.c

```diff
@@ -125,8 +129,14 @@ static void do_boot(struct boot_rsp *rsp)
 #ifdef CONFIG_USB
     /* Disable the USB to prevent it from firing interrupts */
     usb_disable();
+#endif
+#if CONFIG_CLEANUP_ARM_CORE
+    cleanup_arm_nvic(); /* cleanup NVIC registers */
 #endif
     __set_MSP(vt->msp);
+#if CONFIG_CLEANUP_ARM_CORE
+    __set_CONTROL(0x00); /* application will configures core on its own */
```

Comment:
Is there a reason why this is not in a function inside the cleanup_arm module?

Reply:
Better to execute mcuboot call `cleanup_arm_nvic()` without so crucial reconfiguration of the core.
So, This is done intentionally after MSP have been set. Setting control CPU register is something which should be very last operation before jump to the application.

Reply:
makes sense.

Reply:
MSR CONTROL needs to be followed by ISB - that's a recommendation from ARM, in particular if it writes to the SPSEL.

Reply:
addressed in https://github.com/JuulLabs-OSS/mcuboot/pull/856